tests/cpp/test_learner.cc

```diff
@@ -176,6 +177,52 @@ TEST(Learner, JsonModelIO) {
   }
 }
 
+// Crashes the test runner if there are race condiditions.
+//
+// Build with additional cmake flags to enable thread sanitizer
+// which definitely catches problems. Note that OpenMP needs to be
+// disabled, otherwise thread sanitizer will also report false
+// positives.
+//
+// ```
+// -DCMAKE_C_FLAGS="-fsanitize=thread -g -O1"
```

Comment:
XGBoost comes with tsan support in CMake. `-DUSE_SANITIZER=ON`, `-DENABLED_SANITIZERS=thread`

Reply:
Thanks. Adjusted the docs.